frontend/model/contracts/shared/nativeNotification.js

```diff
@@ -1,28 +1,33 @@
 'use strict'
+import sbp from '@sbp/sbp'
 
 // NOTE: since these functions don't modify contract state, it should
 //       be safe to modify them without worrying about version conflicts.
 
 export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
   if (typeof Notification === 'undefined' ||
-    (!force && Notification.permission !== 'default') ||
-    (force && Notification.permission === 'granted')) {
+    (!force && Notification.permission !== 'default')) {
```

Comment:
Why are we returning `null` in this case instead of just calling `await Notification.requestPermission()` below?

What if `Notification.permission` is === `'granted'` here? It would return `null` but we want this to return `'granted'` in that case...

Reply:
> Why are we returning null in this case instead of just calling await Notification.requestPermission() below?

We don't need to request permission because Notification is undefined or user already reacted.
Since the `force` is false, we don't need to request if he is already replied.

Reply:
@Silver-IT The problem though is that it's not specified what this function is supposed to return, and it should be specified.

In some places in the code you check the return value of this function and expect it to be a string, and check to see if it's `'granted'`. In other places you don't check the return the value - but if you're checking for `'granted'` even in a single place - then it's implied that this function returns `'granted'` when permissions are granted.

Yet here the function will return `null` in some cases when permissions are `'granted'`. That is very confusing and I think will lead to bugs.

Reply:
> What if Notification.permission is === 'granted' here? It would return null but we want this to return 'granted' in that case...

The main purpose of this condition is to skip when Notification is `undefined` or user has already replied.
Also it's good idea to return permission if Notification is not `undefined`.

Reply:
OK, then the behavior should be changed to return `null` when `Notification` is `undefined`, and the permission otherwise.

Reply:
```
export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
  if (typeof Notification === 'undefined') {
    return null
  } else if (!force && Notification.permission !== 'default') {
    return Notification.permission
  }
  try {
    return await Notification.requestPermission()
  } catch (e) {
    return null
  }
}
```

So, is this fine with you?

Reply:
How about?

```js
export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
  if (typeof Notification === 'undefined') {
    return null
  }
  if (force) {
    try {
      return await Notification.requestPermission()
    } catch (e) {
      console.error('requestNotificationPermission:', e.message)
      return null
    }
  }
  return Notification.permission
}
```

Reply:
Much better.

Reply:
But your code missed one situation. When force is false and the permission is `default`.

Reply:
```
export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
  if (typeof Notification === 'undefined') {
    return null
  }
  if (force || Notification.permission === 'default') {
    try {
      return await Notification.requestPermission()
    } catch (e) {
      console.error('requestNotificationPermission:', e.message)
      return null
    }
  }
  return Notification.permission
}
```

This could be perfect.

Reply:
> But your code missed one situation. When force is false and the permission is default.

Good catch!!

Is this right?

```js
export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
  if (typeof Notification === 'undefined') {
    return null
  }
  if (force ||  Notification.permission === 'default') {
    try {
      return await Notification.requestPermission()
    } catch (e) {
      console.error('requestNotificationPermission:', e.message)
      return null
    }
  }
  return Notification.permission
}
```

Reply:
> This could be perfect.

We both wrote the same thing 😄

Reply:
Hahaha. Sure. Not only right. It's perfect.